toolchain/check/facet_type.h

```diff
@@ -0,0 +1,44 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_CHECK_FACET_TYPE_H_
+#define CARBON_TOOLCHAIN_CHECK_FACET_TYPE_H_
+
+#include "toolchain/check/context.h"
+#include "toolchain/sem_ir/ids.h"
+
+namespace Carbon::Check {
+
+// Create a FacetType typed instruction object consisting of a single
+// interface.
+auto FacetTypeFromInterface(Context& context, SemIR::InterfaceId interface_id,
+                            SemIR::SpecificId specific_id) -> SemIR::FacetType;
+
+// Adds and returns an `ImplWitness` instruction (created with location set to
+// `witness_loc_id`) that "`Self` type" of type "facet type" implements
```

Comment:
```suggestion
// `witness_loc_id`) that verifies "`Self` type" of type "facet type" implements
```
or "that witnesses"? or "that shows", or some verb there?

Reply:
Used "shows"